### PR TITLE
Add Batpad 2x4 PCB

### PR DESCRIPTION
**Pull request by qunguyentk**
## Description

I have made a macro pad and want to add it to QMK, This is first time I do it. So sorry if something wrong.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/Yncognito/batpad/batpad.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "batpad.h"
+
+void matrix_init_kb(void) {
+    // put your keyboard start-up code here
+    // runs once when the firmware starts up
+
+    matrix_init_user();
+}
```
Comment (drashna) [inline keyboards/Yncognito/batpad/batpad.c:23]: This isn't being used. 
```suggestion
```

File: keyboards/Yncognito/batpad/batpad.c [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+void led_set_kb(uint8_t usb_led) {
+    // put your keyboard LED indicator (ex: Caps Lock LED) toggling code here
+
+    led_set_user(usb_led);
+}
+
+
+// generated by KBFirmware JSON to QMK Parser
+// https://noroadsleft.github.io/kbf_qmk_converter/
```
Comment (drashna) [inline keyboards/Yncognito/batpad/batpad.c:61]:
```suggestion
```

File: keyboards/Yncognito/batpad/batpad.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT( \
```
Comment (drashna) [inline keyboards/Yncognito/batpad/batpad.h:20]:
```suggestion
#define LAYOUT_ortho_2x4( \

```

File: keyboards/Yncognito/batpad/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
```
Comment (drashna) [inline keyboards/Yncognito/batpad/keymaps/default/keymap.c:19]:
```suggestion
    [0] = LAYOUT_ortho_2x4(

```

File: keyboards/Yncognito/batpad/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        KC_Q,    KC_W,    KC_E,    KC_R,
+        KC_LGUI, KC_SPC,  RGB_TOG, RGB_MOD
+    ),
+    [1] = LAYOUT(
```
Comment (drashna) [inline keyboards/Yncognito/batpad/keymaps/default/keymap.c:23]:
```suggestion
    [1] = LAYOUT_ortho_2x4(
```

File: keyboards/Yncognito/batpad/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        KC_Q,    KC_W,    KC_E,    KC_R,
+        KC_LGUI, KC_SPC,  RGB_TOG, RGB_MOD
+    ),
+    [2] = LAYOUT(
```
Comment (drashna) [inline keyboards/Yncognito/batpad/keymaps/default/keymap.c:27]:
```suggestion
    [2] = LAYOUT_ortho_2x4(
```

File: keyboards/Yncognito/batpad/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
```
Comment (drashna) [inline keyboards/Yncognito/batpad/keymaps/via/keymap.c:19]:
```suggestion
    [0] = LAYOUT_ortho_2x4(
```

File: keyboards/Yncognito/batpad/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+        KC_Q,    KC_W,    KC_E,    KC_R,
+        KC_LGUI, KC_SPC,  RGB_TOG, RGB_MOD
+    ),
+    [1] = LAYOUT(
```
Comment (drashna) [inline keyboards/Yncognito/batpad/keymaps/via/keymap.c:23]:
```suggestion
    [1] = LAYOUT_ortho_2x4(
```

File: keyboards/Yncognito/batpad/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+        KC_Q,    KC_W,    KC_E,    KC_R,
+        KC_LGUI, KC_SPC,  RGB_TOG, RGB_MOD
+    ),
+    [2] = LAYOUT(
```
Comment (drashna) [inline keyboards/Yncognito/batpad/keymaps/via/keymap.c:27]:
```suggestion
    [2] = LAYOUT_ortho_2x4(
```

File: keyboards/Yncognito/batpad/batpad.c [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+	{4	,5	,6	,7	},
+	{0	,1	,2	,3	}
+	},
+	{
+	{0,0},{75,0},{145,0},{224,0},
+	{0,64},{75,64},{145,64},{224,64}
+	},
+	{0xff,0xff,0xff,0xff,
+	 0xff,0xff,0xff,0xff}
+};
```
Comment (drashna) [inline keyboards/Yncognito/batpad/batpad.c:31]:
```suggestion
    {
        {4  ,5  ,6  ,7  },
        {0  ,1  ,2  ,3  }
	},
	{
        {0,0},{75,0},{145,0},{224,0},
        {0,64},{75,64},{145,64},{224,64}
    },
    {
        0xff,0xff,0xff,0xff,
        0xff,0xff,0xff,0xff
    }
};

```

**Comment by qunguyentk** [comment]
> Could you rename the folder so that it's using all lower case characters?
Yes, but that mean I have to create new folder and reupdate file to it? 
Thank you so much!

**Comment by qunguyentk** [comment]
> Could you rename the folder so that it's using all lower case characters?

Yes,...  but how to do it? 
Sorry, I have just stated to use Github.
Thank you so much!

**Comment by drashna** [comment]
No worries.  basically, this:
https://help.github.com/en/github/managing-files-in-a-repository/moving-a-file-to-a-new-location

**Comment by qunguyentk** [comment]
Thank you !!!!

File: keyboards/yncognito/batpad/readme.md [hunk 11]
```diff
@@ -0,0 +1,15 @@
+# batpad
+
+![batpad](https://imgur.com/CZ3m0jh)
```
Comment (fauxpark) [inline keyboards/yncognito/batpad/readme.md:3]:
```suggestion
![batpad](https://i.imgur.com/CZ3m0jh.jpg)
```

File: keyboards/yncognito/batpad/readme.md [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+
+A short description of the keyboard/project
```
Comment (fauxpark) [inline keyboards/yncognito/batpad/readme.md:5]: Please fill this out.

File: keyboards/yncognito/batpad/readme.md [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+
+* Keyboard Maintainer: Yncognito
+* Hardware Supported: Batpad/Atmega32u4
+* Hardware Availability: https://www.facebook.com/Yncognito-112984860434908
+
+Make example for this keyboard (after setting up your build environment):
+
+    make Yncognito/batpad:default
```
Comment (fauxpark) [inline keyboards/yncognito/batpad/readme.md:13]:
```suggestion
    make yncognito/batpad:default
```

File: keyboards/yncognito/batpad/rules.mk [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes             # USB Nkey Rollover
+BACKLIGHT_ENABLE = no        # Enable keyboard backlight functionality
+RGB_MATRIX_ENABLE = WS2812       
+MIDI_ENABLE = no             # MIDI support
+UNICODE_ENABLE = no          # Unicode
+BLUETOOTH_ENABLE = no        # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no            # Audio output on port C6
+FAUXCLICKY_ENABLE = no       # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no          # Enable support for HD44780 based LCDs
```
Comment (fauxpark) [inline keyboards/yncognito/batpad/rules.mk:33]:
```suggestion

```

File: keyboards/yncognito/batpad/readme.md (hunk 12 above)
Comment (qunguyentk) [reply keyboards/yncognito/batpad/readme.md:5]: image
I have already changed it

File: keyboards/yncognito/batpad/batpad.c [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+        {4  ,5  ,6  ,7  },
+        {0  ,1  ,2  ,3  }
+	},
+	{
+        {0,0},{75,0},{145,0},{224,0},
+        {0,64},{75,64},{145,64},{224,64}
+    },
+    {
+        0xff,0xff,0xff,0xff,
+        0xff,0xff,0xff,0xff
```
Comment (zvecr) [inline keyboards/yncognito/batpad/batpad.c:31]:
```suggestion
        0x04,0x04,0x04,0x04,
        0x04,0x04,0x04,0x04
```

File: keyboards/yncognito/batpad/readme.md [hunk 16]
```diff
@@ -0,0 +1,14 @@
+# Batpad
+
+![batpad](https://i.imgur.com/CZ3m0jh.jpg)
```
Comment (noroadsleft) [inline keyboards/yncognito/batpad/readme.md:3]: Swaps this image for a smaller version.

```suggestion
![batpad](https://i.imgur.com/CZ3m0jhl.jpg)
```

File: keyboards/yncognito/batpad/batpad.h [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT_ortho_2x4( \
+
+    K00, K01, K02, K03,   \
+    K10, K11, K12, K13    \
```
Comment (noroadsleft) [inline keyboards/yncognito/batpad/batpad.h:23]:
```suggestion
    K00, K01, K02, K03,   \
    K10, K11, K12, K13    \
```

File: keyboards/yncognito/batpad/info.json [hunk 18]
```diff
@@ -0,0 +1,23 @@
+{
+    "keyboard_name": "batpad",
+    "url": "",
+    "maintainer": "qmk",
+    "width": 4,
+    "height": 2,
+    "layouts": {
+        "LAYOUT": {
```
Comment (noroadsleft) [inline keyboards/yncognito/batpad/info.json:8]:
```suggestion
        "LAYOUT_ortho_2x4": {
```

File: keyboards/yncognito/batpad/keymaps/via/keymap.c [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+    ),
+    [1] = LAYOUT_ortho_2x4(
+        KC_Q,    KC_W,    KC_E,    KC_R,
+        KC_LGUI, KC_SPC,  RGB_TOG, RGB_MOD
+    ),
+    [2] = LAYOUT_ortho_2x4(
+        KC_Q,    KC_W,    KC_E,    KC_R,
+        KC_LGUI, KC_SPC,  RGB_TOG, RGB_MOD
+    ),
+
```
Comment (noroadsleft) [inline keyboards/yncognito/batpad/keymaps/via/keymap.c:31]: VIA requires four layers by default.

```suggestion
    [3] = LAYOUT_ortho_2x4(
        _______, _______, _______, _______,
        _______, _______, _______, _______
    ),
```

**Comment by noroadsleft** [comment]
Thanks!

As an aside, we recommend against committing to your `master` branch as you've done here, because pull requests from modified `master` branches can make it more difficult to keep your QMK fork updated. It is highly recommended for QMK development – regardless of what is being done or where – to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your fork updated here:

[**Best Practices: Your Fork's Master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

[Fixing Your Branch](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) will walk you through fixing up your `master` branch moving forward. If you need any help with this just ask.

Thanks again for contributing!

**Pull request merged**